File: frontend/app/types/concept.ts
export interface Concept {
  id: string;
  code: string;
  name: string;
  description?: string;
  concept_type: string;
  debit: string;
  debit_account_id: string;
  debit_account_name?: string; // ✅ Nuevo campo
  credit: string;
  credit_account_id: string;
  credit_account_name?: string; // ✅ Nuevo campo
  active: boolean;
  user_id?: string;
  created_at?: string;
  updated_at?: string;
}

export interface ConceptReadDTO extends Concept {}
export interface ConceptListResponseDTO {
  total: number;
  items: Concept[];
  concept_type_enum: Record<string, string>;
}

export interface CreateConceptData {
  code: string;
  name: string;
  description?: string;
  concept_type: string;
  debit_account_id: string;
  credit: string;
  credit_account_id: string;
  active?: boolean;
}

export interface UpdateConceptData extends Partial<CreateConceptData> {
  id: string;
}

export function mapConcept(dto: ConceptReadDTO): Concept {
  return {
    id: dto.id,
    name: dto.name,
    code: dto.code,
    description: dto.description,
    concept_type: dto.concept_type,
    debit: dto.debit,
    debit_account_id: dto.debit_account_id,
    debit_account_name: dto.debit_account_name, // ✅ Mapear nuevo campo
    credit: dto.credit,
    credit_account_id: dto.credit_account_id,
    credit_account_name: dto.credit_account_name, // ✅ Mapear nuevo campo
    active: dto.active,
    user_id: dto.user_id,
    created_at: dto.created_at,
    updated_at: dto.updated_at,
  };
}

File: backend/app/models/concept.py
import uuid
import enum
from sqlalchemy import Enum as SAEnum
from typing import Optional
from sqlalchemy import String, Boolean, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.dialects.postgresql import UUID as PG_UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.db.base import Base

class ConceptTypeEnum(str, enum.Enum):
    E = "Entrada"
    S = "Salida"
    N = "Neutral"

class Concept(Base):
    __tablename__ = "concepts"

    id: Mapped[uuid.UUID] = mapped_column(PG_UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    code: Mapped[str] = mapped_column(String, index=True, unique=True, nullable=False)
    name: Mapped[str] = mapped_column(String(100), unique=True, nullable=False)
    description: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    
    concept_type: Mapped[ConceptTypeEnum] = mapped_column(
        SAEnum(ConceptTypeEnum, name="concept_type_enum", native_enum=True, validate_strings=True),
        nullable=False,
    )
    debit_account_id: Mapped[uuid.UUID] = mapped_column(PG_UUID(as_uuid=True), ForeignKey("accounts.id"), nullable=False)
    credit_account_id: Mapped[uuid.UUID] = mapped_column(PG_UUID(as_uuid=True), ForeignKey("accounts.id"), nullable=False)
    user_id: Mapped[uuid.UUID] = mapped_column(PG_UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    active: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[DateTime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at: Mapped[DateTime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=True)
    
    # ✅ Relaciones para obtener nombres de cuentas
    debit_account = relationship("Account", foreign_keys=[debit_account_id], lazy="joined")
    credit_account = relationship("Account", foreign_keys=[credit_account_id], lazy="joined")